src/esockd_proxy_protocol.erl

```diff
@@ -98,6 +90,31 @@ recv(Transport, Sock, Timeout) ->
             {error, proxy_proto_timeout}
     end.
 
+recv_v2(Transport, Sock, OriginOpts, Deadline) ->
+    Transport:setopts(Sock, [{active, false}, {packet, raw}]),
+    with_remaining_timeout(Deadline, fun(HeaderTimeout) ->
+        case Transport:recv(Sock, 14, HeaderTimeout) of
+            {ok, <<?SIG, 2:4, Cmd:4, AF:4, Trans:4, Len:16>>} ->
+                with_remaining_timeout(Deadline, fun(ProxyInfoTimeout) ->
```

Comment:
Nit: maybe move the inner fn to a separate top level one since it's really nested here.